annotation-processor/src/main/java/org/fulib/fx/ControllerAnnotationProcessor.java

```diff
@@ -277,4 +288,90 @@ private boolean isProvider(TypeMirror typeMirror) {
     }
 
+    /**
+     * Checks if the given method overrides another event method.
+     *
+     * @param method The method to check
+     */
+    private void checkOverrides(ExecutableElement method) {
+        TypeMirror clazz = method.getEnclosingElement().asType();
+        TypeElement element = (TypeElement) processingEnv.getTypeUtils().asElement(clazz);
+        TypeMirror parentClazz = element.getSuperclass();
+
+        // If no parent class is found, the method cannot override anything
+        if (parentClazz.getKind() == TypeKind.NONE) {
+            return;
+        }
+
+        TypeElement parentElement = (TypeElement) processingEnv.getTypeUtils().asElement(parentClazz);
+
+        streamAllMethods(parentElement)
+                .filter(otherMethod -> otherMethod.getSimpleName().equals(method.getSimpleName()))
+                .filter(this::isEventMethod)
+                .filter(otherMethod -> processingEnv.getTypeUtils().isSubsignature((ExecutableType) method.asType(), (ExecutableType) otherMethod.asType()))
+                .findFirst()
+                .ifPresent(overriddenMethod -> processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, error(1013).formatted(method, element.getQualifiedName(), ((TypeElement) overriddenMethod.getEnclosingElement()).getQualifiedName()), method));
+    }
+
+    private boolean isEventMethod(ExecutableElement method) {
+        for (Class<? extends Annotation> eventAnnotation : ControllerUtil.EVENT_ANNOTATIONS) {
+            if (method.getAnnotation(eventAnnotation) != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    Stream<ExecutableElement> streamMethods(TypeElement componentClass) {
+        return componentClass
+                .getEnclosedElements()
+                .stream()
+                .filter(element -> element instanceof ExecutableElement)
+                .map(element -> (ExecutableElement) element);
+    }
+
+    Stream<ExecutableElement> streamAllMethods(TypeElement componentClass) {
+        return streamSuperClasses(componentClass).flatMap(this::streamMethods);
+    }
+
+    Stream<TypeElement> streamSuperClasses(TypeElement componentClass) {
+        return Stream.iterate(componentClass, Objects::nonNull, e -> (TypeElement) processingEnv.getTypeUtils().asElement(e.getSuperclass()));
+    }
+    // This will throw an error if the methods are private
+    Stream<ExecutableElement> streamAllMethods(TypeElement componentClass, Class<? extends Annotation> annotation) {
+        return streamSuperClasses(componentClass).flatMap(e -> streamMethods(e, annotation)).peek(method -> {
+            if (method.getModifiers().contains(Modifier.PRIVATE)) {
+                processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, error(1012).formatted(Method.class.getSimpleName(), method.getSimpleName(), componentClass.getQualifiedName(), annotation.getSimpleName(), method));
```

Comment:
Will this not potentially report the same error multiple times? I would keep this a pure (as in, no side effects) helper method and do the reporting elsewhere.